pkg/client/client.go

```diff
@@ -34,8 +34,7 @@ type GRPCClient struct {
 }
 
 // NewClient creates a new client for the services for the given address, with the certificate and hmac.
-func NewClient(ctx context.Context, hostname string, port int, certFile string, keyFile string, caFile string, hmacKey string, insecure bool, logger *slog.Logger) (Client, error) {
-
+func NewClient(ctx context.Context, hostname string, port int, certFile string, keyFile string, caFile string, hmacKey string, insecure bool, logger *slog.Logger, namespace string) (Client, error) {
```

Comment:
maybe now we reached the point where we should introduce a configuration struct where optional fields are really optional

Reply:
It's breaking but looks much better. Also we identified one arg that was not even used anymore.